### PR TITLE
C Code generation

### PR DESCRIPTION
**Pull request by isuruf**
Minimum codegen needed for pycalphad.

cc @certik, @richardotis, @asmeurer

### CODE REVIEW

**Comment by isuruf** [comment]
@richardotis, in the benchmark we talked about, 
Here are the timings,

Step | SymEngine | SymPy | Speedup
--- | --- | --- | ---
ccode generation with sympify | 0.0969 | 0.1583 | 1.63x
ccode generation without sympify | 0.0138 | 0.1582 | 11.46x
diff + ccode generation with sympify | 0.0415 | 1.1711 | 28.22x

**Comment by richardotis** [comment]
@bjodah may be interested in this since he wrote https://github.com/bjodah/symcxx

File: symengine/codegen.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    }
+    void bvisit(const UniversalSet &x) {
+        throw std::runtime_error("Not supported");
+    }
+    void _print_pow(std::ostringstream &o, const RCP<const Basic> &a,
+                                const RCP<const Basic> &b)
+    {
+        if (eq(*b, *rational(1, 2))) {
+            o << "sqrt(" << apply(a) << ")";
+        } else {
```
Comment (bjodah) [inline symengine/codegen.h]: Are we targeting C99?
Then we can add `cbrt` as well.

**Comment by bjodah** [comment]
Ah.. missed the minimal part. Nice to see codegen, but I'm a bit surprised that the actual code generation is ever a bottleneck though.

**Comment by certik** [comment]
+1 to merge if tests pass. We can improve upon it in master, this lays a good foundation.

**Pull request merged**